packages/utils/src/payment/abstract-payment-provider.ts

```diff
@@ -0,0 +1,146 @@
+import {
+  IPaymentProcessor,
+  MedusaContainer,
+  PaymentProcessorContext,
+  PaymentProcessorError,
+  PaymentProcessorSessionResponse,
+  PaymentSessionStatus,
+} from "@medusajs/types"
+
+export abstract class AbstractPaymentProcessor implements IPaymentProcessor {
```

Comment:
thought(non-blocking): Should we move back to `AbstractPaymentProvider` and `IPaymentProvider`, as we have had previously? I feel like there is something off about having processor and provider as two separate concepts.

Reply:
If we, at some point, want to introduce a provider for checkout, we can introduce a CheckoutProvider.

Reply:
Yes 👍 I will rename